runtime/commonMain/src/kotlinx/serialization/json/Json.kt

```diff
@@ -122,7 +122,7 @@ public constructor(
      */
     public override fun <T> parse(deserializer: DeserializationStrategy<T>, string: String): T {
         val reader = JsonReader(string)
-        val input = StreamingJsonInput(this, WriteMode.OBJ, reader)
+        val input = StreamingJsonInput(this, WriteMode.OBJ, reader, DescriptorSchemaCache())
```

Comment:
Even if some class is encountered only once in a user's code, caching can still be useful, e.g. if this class is used in a collection.

Schema cache bound to the `Json` instance indeed seems to be a more efficient way, but it will make `Json` a stateful class. We should carefully think about whether we can allow such change or not.

Reply:
> but it will make Json a stateful class

It won't. A cache is an optimization that does not add an observable "stateful" behaviour.